### DISCUSSION TITLE
Flow over a flat plate at an AoA

### DISCUSSION THREAD

**Discussion by gauravity30** [xcompact3d/Incompact3d#241] [Q&A] [answered]
Hello, 
I am using xcompact3d to simulate the flow over flat plate at an angle of attack of 5 degrees at Re=10000. This is for the purpose of validation. I have modified the `bccylinder.f90` file to input the geometry of the flat plate. 
The 2.5% t/c flat plate used for the simulation.
![image](https://github.com/xcompact3d/Incompact3d/assets/53783610/ebed0483-e3cf-4d52-b09d-6073d6c7a949)

Problem: I am neither getting the correct values of drag/lift coefficients nor the correct value of St number. Any suggestions/improvements to get the correct values ?

The parameters for simulation are as follows,
```
p_row=20 !20             ! Row partition
p_col=10 !10              ! Column partition

! Mesh
nx=321               ! X-direction nodes
ny=321                 ! Y-direction nodes
nz=160                ! Z-direction nodes
istret = 1            ! y mesh refinement (0:no, 1:center, 2:both sides, 3:bottom)
beta = 0.5    ! Refinement parameter (beta)

! Domain
xlx = 4      ! Lx (Size of the box in x-direction)
yly = 4            ! Ly (Size of the box in y-direction)
zlz = 2            ! Lz (Size of the box in z-direction)

! Boundary conditions
nclx1 = 2
nclxn = 2
ncly1 = 1
nclyn = 1
nclz1 = 0
nclzn = 0


! Flow parameters
iin = 0               ! Inflow conditions (1: classic, 2: turbinit)
re  = 1000 ! nu=1/re (Kinematic Viscosity)
u1  = 20              ! u1 (max velocity) (for inflow condition)
u2  = 20              ! u2 (min velocity) (for inflow condition)
init_noise  = 0.005   ! Turbulence intensity (1=100%) !! Initial condition
inflow_noise = 0.0    ! Turbulence intensity (1=100%) !! Inflow condition

! Time stepping
dt = 0.00001              ! Time step
ifirst = 0            ! First iteration
ilast = 500001     ! Last iteration

! Enable modelling tools
ilesmod=0             ! if 0 then DNS
iscalar=0             ! If iscalar=0 (no scalar), if iscalar=1 (scalar)
iibm=2                ! Flag for immersed boundary method
```
Thanks,
Gaurav

**Comment by slaizet** [comment]
Dear Gaurav,
What is the size of your control volume for the computation of the lift and drag? It should include the plate. 
Best.
Sylvain

**Reply by gauravity30** [reply]
Yeah, It includes the flat plate. The plate is centered at 1.5, 2 and the control volume boundaries are,
```
xld(1) = 0.8 !X left for volume control
xrd(1) = 2.2 !X right for volume control
yld(1) = 1.5 !Y bottom for volume control
yud(1) = 2.5 !Y top for volume control
```

**Reply by slaizet** [reply, marked as answer]
I guess you resolution is quite coarse so I would aim to increase the resolution if possible.
Other than that, I do not see any issue with your set-up.
Best,
Sylvain

**Reply by gauravity30** [reply]
Okay sir ! I will try with a finer mesh.  Also sir, I have one more question regarding the mesh refinement factor.
What does the mesh refinement factor (Beta) actually represent, is it a ratio of something ? I wanted to know about the mathematical relation which determines the refinement in the y-direction.
Thank You !

**Reply by slaizet** [reply]
You will find all the details in Laizet, S., & Lamballais, E. (2009). High-order compact schemes for incompressible flows: A simple and efficient method with quasi-spectral accuracy. Journal of Computational Physics, 228(16), 5989-6015.

The metric is quite complicated :-)

Best,

Sylvain

**Reply by gauravity30** [reply]
Okay sir. I will look into it. Thank you !

**Reply by gauravity30** [reply]
Merry Christmas Sir ! 
Sir, I am trying to implement another approach here i.e. by rotating the input velocity instead of the geometry. Thus, making the flat plate perfectly in the domain . 

I modified the inflow subroutine of the `BC-Cylinder.f90` file and did the following changes to implement the approach,

```
    A = 5.0 

    ca = 0.996194698 

    sa = 0.087155743 


    do k=1,xsize(3)

       do j=1,xsize(2)

          bxx1(j,k)=u1*ca+bxo(j,k)*inflow_noise

          bxy1(j,k)=u1*sa+byo(j,k)*inflow_noise

          bxz1(j,k)=zero+bzo(j,k)*inflow_noise

       enddo

    enddo

```

The error in drag and lift was comparatively more than the approach where we rotate the geometry also the flow was different. 

https://github.com/xcompact3d/Incompact3d/assets/53783610/a02a0acc-aa2a-4af9-8e41-2e44fd922894

I feel I might have missed some modifications to correctly implement the approach. Can you help regd. the same ?

Thank You.

**Reply by slaizet** [reply]
Hi,
If you change the inflow direction, you will have to change the BC everywhere accordingly (you would have fluid going out of the computational domain not just at the outflow BC. I think it is a quite tricky set-up.
Best,
Sylvain
